### PR TITLE
fix: pr#120

### PR DESCRIPTION
**Pull request by uki00a**

### CODE REVIEW

File: tests/test_util.ts [hunk 1]
```diff
@@ -23,29 +22,25 @@ export class TestSuite {
   }
 
   test(name: string, func: TestFunc): void {
-    this.tests.push({ name, func });
+    Deno.test(`[${this.prefix}] ${name}`, async () => {
```
Comment (uki00a) [inline tests/test_util.ts:25]: We want to utilize Deno's leaking check feature :smiley:.
So, I run `func` here.

**Comment by tumile** [comment]
Is it assured that `afterAlls` are always run after the tests? From the CI check it seems like all working fine, but I'm not sure if Deno tests are always executed in the same order as declared?

**Comment by uki00a** [comment]
> Deno tests are always executed in the same order as declared?

Yes, it is.

**Comment by tumile** [comment]
Awesome. I feel like this could work too

```ts
runTests = (): void => {
  for (const test of this.tests) {
    Deno.test(`[${this.prefix}] ${test.name}`, async () => {
      for (const f of this.beforeEachs) {
        await f();
      }
      await test.func();
    });
  }
  Deno.test({
    name: `[${this.prefix}] afterAll`,
    fn: async () => {
      for (const f of this.afterAlls) {
        await f();
      }
    },
    sanitizeOps: false,
    sanitizeResources: false,
  });
};
```

I want to keep the tests array to defer the execution of tests till the end so the call to `beforeEach` can be anywhere in the file but not necessarily up top

**Comment by tumile** [comment]
I can update my PR directly with this. Need to remove all `await` in `await suite.runTests()` also since it's not async anymore

**Pull request closed without merging**